### PR TITLE
#2: [YOLOv1] data_collate function

### PR DESCRIPTION
**Pull request by insurgent92**

### CODE REVIEW

File: posts/03_01_dataloader.md [hunk 1]
```diff
@@ -320,5 +397,79 @@ class VOC(data.Dataset):
...
+```
+
+- 정의한 VOC class의 인스턴스를 생성합니다.
+- 생성된 인스턴스를 `torch.utils.data.DataLoader`의 인자로 줍니다.
+
+​    
+
+### 1. collate_fn
+
+  앞서 Dataloader를 살펴보았습니다. YOLOv1의 output tensor의 형태는 S x S x (B * 5 + C) 입니다. Loss를 계산하고 학습하려면 label을 output tensor와 동일한 형태로 구성해야 합니다. 이를 위해서 PyTorch에서는 `torch.utils.data.DataLoader`에서 `collate_fn`라는 파라미터를 이용할 수 있습니다. (`torch.utils.data.DataLoader`에서 제공하는 모든 파라미터는 [이곳](https://pytorch.org/docs/stable/data.html?highlight=dataloader)을 참고하시기 바랍니다)
```
Comment (ssaru) [inline posts/03_01_dataloader.md]: 문서 작성을 완료하고 하면 문서 통일성 확인을 전반적으로 검토해야할 것 같습니다. ㅎㅎ

File: posts/03_01_dataloader.md [hunk 2]
```diff
@@ -320,5 +397,79 @@ class VOC(data.Dataset):
...
+
+            np_label[grid_x_index][grid_y_index] = np.array([objectness, x_offset, y_offset, w_ratio, h_ratio, classes])
+
+        label = torch.from_numpy(np_label)
+        targets.append(label)
+    return torch.stack(imgs, 0), torch.stack(targets, 0), sizes
+```
+`detection_collate`의 입력 파라미터인 batch는 list로 ``batch[0]``는 이미지, ``batch[1]``는 레이블, ``batch[3]``는 원본 이미지 사이즈입니다. 
+#### Normalized bounding box  
+``batch[1]``는 레이블 정보인 `[x, y, w, h, class]`를 list로 담고 있습니다. output tensor에는 정규화된 bounding box 좌표를 사용합니다. bounding box의 중심점인 (x, y)는 각 그리드 셀에서의 상대적인 위치로 나타내며, 0-1 범위의 값을 가집니다. bounding box의 너비와 높이인 (w, h)는 전체 이미지에 대한 상대적인 위치를 나타내며, 0-1의 범위를 가집니다.
```
Comment (ssaru) [inline posts/03_01_dataloader.md]: 해당 설명은 그림이 같이 포함되면 이해하는데 더 쉬울 것 같습니다.
